src/controllers/users.js

```diff
@@ -21,29 +20,22 @@ export async function registerUser(req, res) {
     if (!validateEmail(user.email)) {
       throw new Error("Invalid email format");
     }
-
-    const salt = await bcrypt.genSalt(10);
-    const hash = await bcrypt.hash(user.password, salt);
-
-    const newUser = await UserModel.create({
-      username: user.username,
-      email: user.email,
-      hash: hash,
-    });
-    const token = signToken(newUser.get());
-    newUser.setDataValue("token", token);
-    await newUser.save();
-    const responseData = userResponse(newUser.get());
+    const { password, ...tokenPayload } = user;
+    const token = signToken(tokenPayload);
+    const newUser = await createUser(user);
+    const responseData = userResponse(newUser, token);
     return res.status(201).json(responseData);
   } catch (e) {
     console.error(e);
-    if (e.message === "No payload found")
-      return res.status(400).send(e.message);
-    if (e.message === "Invalid payload format")
-      return res.status(400).send(e.message);
-    if (e.message === "Invalid email format")
+    if (
+      e.message === "No payload found" ||
```

Comment:
nit: we can move the error messages into an array or object and just check if e.message is in it. Should make it cleaner and easier to change in the future.